Sources/tart/OCI/Layerizer/DiskV2.swift

```diff
@@ -152,7 +169,25 @@ class DiskV2: Disk {
     var offset = offset
 
     for chunk in data.chunks(ofCount: holeGranularityBytes) {
-      // Only write chunks that are not zero
+      // If the local layer cache is used, only write chunks that differ
+      // since the base disk can contain anything at any position
+      if let rdisk = rdisk {
+        try rdisk.seek(toOffset: offset)
+        let actualContentsOnDisk = try rdisk.read(upToCount: chunk.count)
+
+        if chunk != actualContentsOnDisk {
+          try disk.seek(toOffset: offset)
+          disk.write(chunk)
```

Comment:
It seems we can validate that the new chunk are zeroes here and use `g` syscall to erase a range. Here is an example that might work:

https://github.com/niw/HolePunch/blob/0f6716324915ddf073acb37e1dc268f38b071d3e/Sources/HolePunch/Command.swift#L76-L84

Reply:
Implemented in https://github.com/cirruslabs/tart/pull/864/commits/c110158b6ce7e93c21c35d16dbd564543f5eba8b.